src/middlewares/error.ts

```diff
@@ -11,30 +12,34 @@ const zodErrorOptions: ErrorMessageOptions = {
   transform: ({ errorMessage, index }) => `Error #${index + 1}: ${errorMessage}`
 }
 
-const errorConverter = (err: any, sentry: Toucan) => {
+export const errorConverter = (err: unknown, sentry: Toucan): ApiError => {
   let error = err
   if (error instanceof ZodError) {
     const errorMessage = generateErrorMessage(error.issues, zodErrorOptions)
     error = new ApiError(httpStatus.BAD_REQUEST, errorMessage)
   } else if (!(error instanceof ApiError)) {
-    const statusCode = error.statusCode || httpStatus.INTERNAL_SERVER_ERROR
-    const message = error.description || error.message || httpStatus[statusCode]
+    const castedErr = (typeof error === 'object' ? error : {}) as Record<string, unknown>
+    const statusCode: number =
+      typeof castedErr.statusCode === 'number'
+        ? castedErr.statusCode
+        : httpStatus.INTERNAL_SERVER_ERROR
+    const message = (castedErr.description || castedErr.message || httpStatus[statusCode]) as string
     if (statusCode >= 500) {
       // Log any unhandled application error
       sentry.captureException(error)
     }
-    error = new ApiError(statusCode, message, false, err.stack)
```

Comment:
Because the `ApiError` `constructor` only accepts 3 arguments:
`constructor(statusCode: number, message: string, isOperational = true)`
so the `err.stack` wasn't doing anything and it was giving type errors